packages/aws-client/S3.js

```diff
@@ -594,20 +657,24 @@ exports.validateS3ObjectChecksum = async ({
 };
 
 /**
- * Extract the S3 bucket and key from the URL path parameters
+ * Extracts the S3 bucket and key from the URL path parameters.
  *
  * @param {string} pathParams - path parameters from the URL
- * @returns {Object} - bucket/key in the form of
- * { Bucket: x, Key: y }
+ * @returns {Array<string>} 2-element array containing the bucket and the key,
+ *    respectively
  */
 exports.getFileBucketAndKey = (pathParams) => {
-  const fields = pathParams.split('/');
-
+  const fields = (pathParams.startsWith('/')
+    ? pathParams.slice(1) // Remove leading slash
```

Comment:
Consider: Bugfix or behavior change requiring CHANGELOG update?      I can't imagine the leading slash is a good thing...